### PR TITLE
feat: filter date group

### PR DESCRIPTION
**Pull request by ybzky**
close #xxx

## Pull Request Checklist

- [x] Related tickets or issues have been linked in the PR description (or missing issue).
- [x] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [x] Unit tests have been added for the changes (if applicable).
- [x] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

File: packages/design/src/components/tree/Tree.tsx [hunk 1]
```diff
@@ -34,54 +35,64 @@ export interface ITreeNodeProps {
     children?: ITreeNodeProps[];
 }
 
+export interface IAttachTreeProps {
+    level?: number;
+    count?: number;
+}
+
+type TreeItemProps = ITreeNodeProps & IAttachTreeProps;
```
Comment (wzhudev) [inline packages/design/src/components/tree/Tree.tsx:43]: Should start with an `I` as prefix.

File: packages/sheets-filter-ui/src/locale/en-US.ts [hunk 2]
```diff
@@ -67,6 +67,20 @@ const locale: typeof zhCN = {
         msg: {
             'filter-header-forbidden': 'You can\'t move the header row of a filter.',
         },
+        date: {
```
Comment (wzhudev) [inline packages/sheets-filter-ui/src/locale/en-US.ts:70]: In the future we need to support countries that do not use Gregorian calendar.

File: packages/sheets-filter-ui/src/worker/generate-filter-values.service.ts [hunk 3]
```diff
@@ -137,3 +133,164 @@ export function getFilterByValueItems(
 
     return items;
 }
+
+export function getFilterTreeByValueItems(
```
Comment (wzhudev) [inline packages/sheets-filter-ui/src/worker/generate-filter-values.service.ts:137]: This should have test covered. For example, different number formats of cells.

**Review by wzhudev — CHANGES_REQUESTED** [review]

File: packages/sheets-filter-ui/src/worker/generate-filter-values.service.ts (hunk 3 above)
Comment (ybzky) [reply packages/sheets-filter-ui/src/worker/generate-filter-values.service.ts:138]: different types of formats are added to the filter test demo data

**Review by wzhudev — APPROVED** [review]

**Pull request merged**